default/scripting/techs/defense/mines.macros

```diff
@@ -271,3 +190,17 @@ EG_SYSTEM_MINES
                     empire = Target.Owner
             ]
 '''
+
+SYSTEM_CONTAINS_SOURCE_OR_OWNED_PLANET
+'''                ContainedBy And [
+                    System 
+                    Contains Or [
+                        Source
+                        And [
+                            Planet
+                            OwnedBy empire = Source.Owner
+                            OwnerHasTech name = ThisTech
```

Comment:
How does a has-tech condition affect whether something applies empire-wide? Either an empire has a tech or it doesn't; there's no distinction about this from some of the things the empire owns.